webapp/src/Controller/API/ContestController.php

```diff
@@ -325,62 +325,109 @@ public function setBannerAction(Request $request, string $cid, ValidatorInterfac
      *                 type="boolean",
+     *             ),
+     *             @OA\Property(
+     *                 property="scoreboard_thaw_time",
+     *                 description="The new unfreeze (thaw) time of the contest",
+     *                 type="string",
+     *                 format="date-time"
      *             )
      *         )
      *     )
      * )
      * @OA\Response(
      *     response="200",
-     *     description="Contest start time changed successfully",
+     *     description="Contest object if it changed",
      *     @OA\JsonContent(
-     *         type="string"
+     *         allOf={
+     *             @OA\Schema(ref=@Model(type=Contest::class)),
+     *             @OA\Schema(ref="#/components/schemas/Banner")
+     *         }
      *     )
      * )
+     * @OA\Response(
+     *     response="204",
+     *     description="The change was successful"
+     * )
      */
-    public function changeStartTimeAction(Request $request, string $cid): Response
+    public function changeTimesAction(Request $request, string $cid): Response
     {
         $contest  = $this->getContestWithId($request, $cid);
-        $now      = Utils::now();
+        $now      = (int)Utils::now();
         $changed  = false;
         if (!$request->request->has('id')) {
-            $response = new JsonResponse('Missing "id" in request.', Response::HTTP_BAD_REQUEST);
-        } elseif (!$request->request->has('start_time')) {
-            $response = new JsonResponse('Missing "start_time" in request.', Response::HTTP_BAD_REQUEST);
-        } elseif ($request->request->get('id') != $contest->getApiId($this->eventLogService)) {
-            $response = new JsonResponse('Invalid "id" in request.', Response::HTTP_BAD_REQUEST);
-        } elseif (!$request->request->has('force') &&
-            $contest->getStarttime() != null &&
-            $contest->getStarttime() < $now + 30) {
-            $response = new JsonResponse('Current contest already started or about to start.',
-                                         Response::HTTP_FORBIDDEN);
-        } elseif ($request->request->get('start_time') === null) {
-            $this->em->persist($contest);
-            $contest->setStarttimeEnabled(false);
-            $response = new JsonResponse('Contest paused :-/.', Response::HTTP_OK);
-            $this->em->flush();
-            $changed = true;
-        } else {
-            $date = date_create($request->request->get('start_time'));
-            if ($date === false) {
-                $response = new JsonResponse('Invalid "start_time" in request.', Response::HTTP_BAD_REQUEST);
+            throw new BadRequestHttpException('Missing "id" in request.');
+        }
+        if (!$request->request->has('start_time') && !$request->request->has('scoreboard_thaw_time')) {
+            throw new BadRequestHttpException('Missing "start_time" or "scoreboard_thaw_time" in request.');
+        }
```

Comment:
I think the check that the request doesn't have both a `start_time` and `scoreboard_thaw_time` was removed.
This means that now if both are provided, the start time may get updated (and persisted) and then when handling the thaw time, the request may fail with a `BadRequestHttpException`. I think that should not be allowed. So probably the simplest is to not allow updating both.

Reply:
Yep I think it's better to re-add that. I will do that, and add a test that checks for it as well.